lib/publiccloud/provider.pm

```diff
@@ -708,19 +708,31 @@ sub terraform_param_tags
 =head2 get_terraform_output
 
 Query the terraform data structure in json format.
+Automatic search and set valid terraform directory on helper image.
 Input: <jq-query-format> string; <empty> = no query then full output of data structure.
 E.g: to get the VM instance name from json data structure, the call is: 
     get_terraform_output(".vm_name.value[0]");
 To get the complete output structure, the call is:
-    get_terraform_output();
+    get_terraform_output(".");
 
 =cut
 
 sub get_terraform_output {
     my ($self, $jq_query) = @_;
-    my $res = script_output("terraform output -no-color -json | jq -r '$jq_query' 2>/dev/null", proceed_on_failure => 1);
-    # jq 'null' shall return empty
-    return $res unless ($res =~ /^null$/);
+    # cuurent dir saved
+    my $dir = script_output("pwd");
+    # terraform state file search first matching
+    my $cd = script_output("find ~ -name \*.tfstate -exec dirname {} + -quit 2>/dev/null");
+    # state file missing: no data found
+    return "" if ($cd eq "");
```

Comment:
```suggestion
    die('Terraform directory not found) if ($cd eq "");
```

Reply:
I preferred to avoid test flow control in this sub routine, leaving the result empty and managed at the caller upper level.